fix: 테스트 고정 날짜와 파티션 TTL 충돌로 마트 테스트 3건 상시 실패

## Problem / 문제

마트 관련 테스트 3건이 시간이 지나면 반드시 실패한다. 테스트가 대상 날짜를 상수로 박아두는데, 집계기의 파티션 TTL 은 `date.today()` 기준 상대 계산이라 두 값이 어긋나는 순간 방금 만든 파티션이 같은 실행 안에서 DROP 된다.

특정 커밋의 회귀가 아니라 **달력이 지나면서 터지는 선재 결함**이다. 2026-08-21 기준 3건 모두 빨간불이며, PR #103 이전 커밋 `21537e3` 워크트리에서도 동일하게 재현된다.

## Steps to Reproduce / 재현 절차

```bash
PYTHONPATH=. TEST_POSTGRES_PORT=5433 uv run pytest -q \
  tests/test_post_aggregate_validation.py::test_post_aggregate_validation_passes_for_valid_target_date_marts \
  tests/test_backend_datamart_contract.py::test_backend_datamart_readiness_step_generates_semantic_rows \
  tests/test_backend_datamart_serving_readiness.py::test_backend_datamarts_are_populated_by_real_pipeline_cli_aggregate_step
```

## Expected Behavior / 기대 동작

집계가 대상 날짜 파티션에 적재한 행이 그대로 남아 `srv_daily_review_list` 조회와 `mart_freshness` 검증을 통과한다.

## Actual Behavior / 실제 동작

`GoldAggregator.run()` 이 적재 직후 `_drop_old_partitions()` 를 호출하고(`src/gold/aggregator.py:128`, `src/gold/aggregator.py:208`), 그 안에서 `cutoff = date.today() - timedelta(days=retention_days)` 로 경계를 잡는다(`src/gold/aggregator.py:278`, 기본 `retention_days=14`). 테스트의 고정 날짜가 이미 cutoff 보다 과거이므로 방금 쓴 파티션이 DROP 되고, `srv_daily_review_list` 가 0행이 되어 검증이 실패한다.

고정 날짜 위치:

| 파일 | 위치 | 값 |
|---|---|---|
| `tests/test_post_aggregate_validation.py` | `:13`~`:20` | `date(2026, 5, 10)` 외 7개 |
| `tests/test_backend_datamart_serving_readiness.py` | `:22` | `TARGET_DATE = date(2026, 5, 3)` |
| `tests/test_backend_datamart_contract.py` | `:543` | `FOR VALUES FROM ('2026-05-02') TO ('2026-05-03')` |

## Environment / 환경

- Commit/branch: `main`(`21537e3`) 및 `feat/issue-40-dept-assign-wiring` 양쪽 재현
- Relevant config: `GoldAggregator.run(retention_days=14)`

## Logs or Evidence / 로그 또는 근거

```
3 failed, 500 passed, 8 warnings in 9.86s
FAILED tests/test_backend_datamart_contract.py::test_backend_datamart_readiness_step_generates_semantic_rows
FAILED tests/test_backend_datamart_serving_readiness.py::test_backend_datamarts_are_populated_by_real_pipeline_cli_aggregate_step
FAILED tests/test_post_aggregate_validation.py::test_post_aggregate_validation_passes_for_valid_target_date_marts
```

## Acceptance Criteria / 완료 조건

- [ ] 세 테스트가 대상 날짜를 `date.today()` 기준 상대값으로 잡거나, TTL 을 명시적으로 넓혀 시간 경과와 무관하게 통과한다.
- [ ] 파티션 TTL 로직 자체는 그대로 둔다(운영 동작 변경 아님). 테스트 쪽에서 해소한다.
- [ ] 시스템 시각을 미래로 옮겨도(예: +90일) 세 테스트가 통과함을 확인한다.
- [ ] 같은 패턴의 고정 날짜가 다른 마트 테스트에 더 있는지 훑고, 있으면 함께 정리한다.

## Validation / 검증

```bash
PYTHONPATH=. TEST_POSTGRES_PORT=5433 uv run pytest -q tests/test_post_aggregate_validation.py \
  tests/test_backend_datamart_contract.py tests/test_backend_datamart_serving_readiness.py
```

--- a/tests/test_backend_datamart_contract.py
+++ b/tests/test_backend_datamart_contract.py
@@ -7,7 +7,7 @@ or a SQLite fallback.
 
 from __future__ import annotations
 
-from datetime import date, datetime, timezone
+from datetime import date, datetime, timedelta, timezone
 from pathlib import Path
 from uuid import UUID
 
@@ -94,10 +94,17 @@ CONTRACT_TABLES = {
 }
 
 SERVICE_FK_TABLES = set(CONTRACT_TABLES)
-CONTRACT_TARGET_DATE = date(2026, 5, 2)
-CONTRACT_PARTITION_NAME = "srv_daily_review_list_contract_2026_05_02"
-READINESS_TARGET_DATE = date(2026, 5, 3)
-READINESS_PARTITION_NAME = "srv_daily_review_list_2026_05_03"
+# 대상 날짜는 오늘 기준 상대값이어야 한다. aggregate 스텝은 적재 직후
+# date.today() - retention_days(기본 14) 이전 파티션을 DROP 하므로, 고정 날짜는
+# 시간이 지나면 방금 쓴 파티션이 같은 실행 안에서 사라진다.
+# 오프셋 1~13 은 마트 테스트 전체가 나눠 쓴다(파일 간 날짜 충돌 방지).
+_TODAY = date.today()
+READINESS_TARGET_DATE = _TODAY - timedelta(days=2)
+CONTRACT_TARGET_DATE = _TODAY - timedelta(days=3)
+# contract_ 접두 파티션은 TTL 정규식(srv_daily_review_list_\d{4}_\d{2}_\d{2})에
+# 걸리지 않아 자동 DROP 대상이 아니다. 정리는 _cleanup_contract_probe 가 한다.
+CONTRACT_PARTITION_NAME = f"srv_daily_review_list_contract_{CONTRACT_TARGET_DATE:%Y_%m_%d}"
+READINESS_PARTITION_NAME = f"srv_daily_review_list_{READINESS_TARGET_DATE:%Y_%m_%d}"
 CONTRACT_DOC_MARKERS = {
     "## Contract boundary": "contract table list",
     "## Column contract matrices": "column contract section",
@@ -540,7 +547,8 @@ def _create_contract_partition(test_db_session) -> None:
             f"""
             CREATE TABLE IF NOT EXISTS public.{CONTRACT_PARTITION_NAME}
             PARTITION OF public.srv_daily_review_list
-            FOR VALUES FROM ('2026-05-02') TO ('2026-05-03')
+            FOR VALUES FROM ('{CONTRACT_TARGET_DATE.isoformat()}')
+                        TO ('{(CONTRACT_TARGET_DATE + timedelta(days=1)).isoformat()}')
             """
         )
     )
@@ -585,7 +593,14 @@ def _cleanup_contract_probe(test_db_session, *, service_id, review_id) -> None:
 
 
 def _insert_contract_rows(test_db_session, service_id, review_id) -> None:
-    reviewed_at = datetime(2026, 5, 2, 9, 30, tzinfo=timezone.utc)
+    reviewed_at = datetime(
+        CONTRACT_TARGET_DATE.year,
+        CONTRACT_TARGET_DATE.month,
+        CONTRACT_TARGET_DATE.day,
+        9,
+        30,
+        tzinfo=timezone.utc,
+    )
     test_db_session.execute(
         text(
             """
@@ -708,7 +723,14 @@ def _insert_readiness_upstream_rows(
     review_id,
     platform_review_id: str,
 ) -> None:
-    reviewed_at = datetime(2026, 5, 3, 9, 30, tzinfo=timezone.utc)
+    reviewed_at = datetime(
+        READINESS_TARGET_DATE.year,
+        READINESS_TARGET_DATE.month,
+        READINESS_TARGET_DATE.day,
+        9,
+        30,
+        tzinfo=timezone.utc,
+    )
     connection.execute(
         text(
             """

--- a/tests/test_backend_datamart_serving_readiness.py
+++ b/tests/test_backend_datamart_serving_readiness.py
@@ -7,7 +7,7 @@ assert backend-facing mart rows and semantics from PostgreSQL.
 
 from __future__ import annotations
 
-from datetime import date, datetime, timezone
+from datetime import date, datetime, timedelta, timezone
 from pathlib import Path
 
 import pytest
@@ -19,7 +19,10 @@ from uuid6 import uuid7
 from src.pipeline import cli as pipeline_cli
 
 
-TARGET_DATE = date(2026, 5, 3)
+# 오늘 기준 상대값. 파이프라인 CLI aggregate 는 적재 직후 파티션 TTL(기본 14일)을
+# 적용하므로, 고정 날짜는 시간이 지나면 방금 쓴 파티션이 같은 실행 안에서 DROP 된다.
+# 오프셋 1~13 은 마트 테스트 전체가 나눠 쓴다(파일 간 날짜 충돌 방지).
+TARGET_DATE = date.today() - timedelta(days=1)
 SERVICE_NAME = "Serving Readiness Bank"
 PLATFORM_REVIEW_PREFIX = "serving-readiness"
 MANUAL_PROOF_DOC = (
@@ -262,9 +265,10 @@ def _seed_analyzed_pipeline_rows(
     platform_review_ids,
 ) -> None:
     reviewed_at = [
-        datetime(2026, 5, 3, 9, 0, tzinfo=timezone.utc),
-        datetime(2026, 5, 3, 10, 0, tzinfo=timezone.utc),
-        datetime(2026, 5, 3, 11, 0, tzinfo=timezone.utc),
+        datetime(
+            TARGET_DATE.year, TARGET_DATE.month, TARGET_DATE.day, hour, 0, tzinfo=timezone.utc
+        )
+        for hour in (9, 10, 11)
     ]
     session.execute(
         text(

--- a/tests/test_pipeline_cli_validation.py
+++ b/tests/test_pipeline_cli_validation.py
@@ -1,6 +1,6 @@
 from __future__ import annotations
 
-from datetime import date, datetime, timezone
+from datetime import date, datetime, timedelta, timezone
 from uuid import uuid4
 
 import pytest
@@ -9,8 +9,11 @@ from sqlalchemy import text
 from src.pipeline import cli as pipeline_cli
 
 
-CLI_WARNING_TARGET_DATE = date(2026, 5, 18)
-CLI_FAILURE_TARGET_DATE = date(2026, 5, 19)
+# 오늘 기준 상대값. 고정 날짜는 파티션 TTL(기본 14일) 밖으로 밀려나 마트 검증이
+# 시간이 지난 뒤 깨진다. 오프셋 1~13 은 마트 테스트 전체가 나눠 쓴다.
+_TODAY = date.today()
+CLI_WARNING_TARGET_DATE = _TODAY - timedelta(days=12)
+CLI_FAILURE_TARGET_DATE = _TODAY - timedelta(days=13)
 
 
 def _target_datetime(target_date: date) -> datetime:

--- a/tests/test_post_aggregate_validation.py
+++ b/tests/test_post_aggregate_validation.py
@@ -1,7 +1,7 @@
 from __future__ import annotations
 
 import re
-from datetime import date, datetime, timezone
+from datetime import date, datetime, timedelta, timezone
 
 import pytest
 from sqlalchemy import text
@@ -10,14 +10,19 @@ from uuid6 import uuid7
 from src.pipeline.steps import run_aggregate, run_post_aggregate_validation
 
 
-SUCCESS_TARGET_DATE = date(2026, 5, 10)
-WARNING_ONLY_TARGET_DATE = date(2026, 5, 11)
-PENDING_BATCH_TARGET_DATE = date(2026, 5, 12)
-FAILED_REVIEW_TARGET_DATE = date(2026, 5, 13)
-MISSING_MAPPING_TARGET_DATE = date(2026, 5, 14)
-ORPHAN_TARGET_DATE = date(2026, 5, 15)
-MART_MISSING_TARGET_DATE = date(2026, 5, 16)
-COUNT_MISMATCH_TARGET_DATE = date(2026, 5, 17)
+# 대상 날짜는 오늘 기준 상대값이어야 한다. GoldAggregator 는 적재 직후
+# _drop_old_partitions() 로 date.today() - retention_days(기본 14) 이전 파티션을
+# DROP 하므로, 고정 날짜를 쓰면 시간이 지난 뒤 방금 쓴 파티션이 같은 실행 안에서
+# 사라진다. 오프셋은 마트 테스트 전체가 1~13 을 나눠 쓴다(파일 간 날짜 충돌 방지).
+_TODAY = date.today()
+SUCCESS_TARGET_DATE = _TODAY - timedelta(days=4)
+WARNING_ONLY_TARGET_DATE = _TODAY - timedelta(days=5)
+PENDING_BATCH_TARGET_DATE = _TODAY - timedelta(days=6)
+FAILED_REVIEW_TARGET_DATE = _TODAY - timedelta(days=7)
+MISSING_MAPPING_TARGET_DATE = _TODAY - timedelta(days=8)
+ORPHAN_TARGET_DATE = _TODAY - timedelta(days=9)
+MART_MISSING_TARGET_DATE = _TODAY - timedelta(days=10)
+COUNT_MISMATCH_TARGET_DATE = _TODAY - timedelta(days=11)
 
 
 @pytest.mark.requires_db

